src/ast/files.cc

```diff
@@ -35,4 +35,11 @@ namespace files
 
     return data;
   }
+
+  void write(const std::string& file, std::string content)
```

Comment:
Should be `const std::string& content`. This needs error handling in case the file can't be written. Sadly, output filenames can have path components, which means needing to parse them and create intermediate directories if necessary. That seems like too much for this feature - we can either error out there (annoying to the user) or perhaps dump to stdout instead of a file and rely on redirects?